plugins/plugin-chart-echarts/src/Gauge/controlPanel.tsx

```diff
@@ -0,0 +1,285 @@
+          {
+            name: 'split_number',
+            config: {
+              type: 'SliderControl',
+              label: t('Split number'),
+              description: t('Number of split segments on the axis'),
+              renderTrigger: true,
+              min: 3,
+              max: 30,
+              default: DEFAULT_FORM_DATA.splitNumber,
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Progress')}</h1>],
+        [
+          {
+            name: 'show_progress',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Show progress'),
+              description: t('Whether to show the progress of gauge chart'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.showProgress,
+            },
+          },
+        ],
+        [
+          {
+            name: 'overlap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Overlap'),
+              description: t(
+                'Whether the progress bar overlaps when there are multiple groups of data',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.overlap,
+            },
+          },
+        ],
+        [
+          {
+            name: 'roundcap',
+            config: {
+              type: 'CheckboxControl',
+              label: t('Roundcap'),
+              description: t('Style the ends of the progress bar with a roundcap'),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.roundcap,
+            },
+          },
+        ],
+        [<h1 className="section-header">{t('Intervals')}</h1>],
+        [
+          {
+            name: 'intervals',
+            config: {
+              type: 'TextControl',
+              label: t('Interval bounds'),
+              description: t(
+                'Comma-separated interval bounds, e.g. 2,4,5 for intervals 0-2, 2-4 and 4-5. Last number should match the value provided for MAX.',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.intervals,
+            },
+          },
+        ],
+        [
+          {
+            name: 'interval_color_indices',
+            config: {
+              type: 'TextControl',
+              label: t('Interval colors'),
+              description: t(
+                'Comma-separated color picks for the intervals, e.g. 1,2,4. Integers denote colors from the chosen color scheme and are 1-indexed. Length must be matching that of interval bounds.',
+              ),
+              renderTrigger: true,
+              default: DEFAULT_FORM_DATA.intervalColorIndices,
+            },
+          },
+        ],
+      ],
+    },
+  ],
+
+  controlOverrides: {
+    columns: {
+      validators: [validateNonEmpty],
+    },
+    row_limit: {
+      choices: [...Array(10).keys()].map(n => n + 1),
+      default: DEFAULT_FORM_DATA.rowLimit,
+    },
+  },
```

Comment:
Some comments on this section:
- In general I recommend avoiding the `controlOverrides` section and moving the logic directly into the control.
- `columns` is not used in this chart, so that section can be removed.